assert/assertions.go

```diff
@@ -521,30 +527,36 @@ func NotSame(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}
 	if h, ok := t.(tHelper); ok {
 		h.Helper()
 	}
-
-	if samePointers(expected, actual) {
-		return Fail(t, fmt.Sprintf(
-			"Expected and actual point to the same object: %p %#v",
-			expected, expected), msgAndArgs...)
+	if same, ok := samePointers(expected, actual); ok {
```

Comment:
Flip the condition on this if to `!ok` to preserve [line of sight](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88).

Reply:
> Flip the condition on this if to `!ok` to preserve [line of sight](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88).

I've instead split it into 2 different statements to make it more readable and also to preserve line of sight:

```
       same, ok := samePointers(expected, actual)

	if ok {}
```